src/gpio.rs

```diff
@@ -19,6 +19,9 @@ use crate::peripheral::{Peripheral, PeripheralRef};
 
 pub use chip::*;
 
+#[cfg(feature = "nightly")]
+use self::asynch::InputFuture;
```

Comment:
Nit: you can also move `InputFuture` out of the `asynch` module, if you want to keep all of the `asynch` module behind the `nightly` feature. `nightly` there is needed as you are implementing a trait which uses the `async-fn-in-trait` nightly feature.